dask/distributed/ipython_utils.py

```diff
@@ -0,0 +1,34 @@
+from .client import Client
+
+
+def dask_client_from_ipclient(ipclient):
+    """
+    Construct a scheduler from an ipython client.
+
+    1. Run a command on the first engine to start a scheduler. This should
+       return an address for clients and an address for workers.
+    2. Run some start worker command on the rest of the engines, giving them
+       the scheduler_to_engines address.
+    3. Start a client here with the scheduler to client address.
+    """
+
+    def start_scheduler():
+        from dask.distributed import Scheduler
+        scheduler = Scheduler()
+        return scheduler.to_clients, scheduler.to_workers
```

Comment:
You want address_to_clients and address_to_workers.  These are the sockets themselves